src/Compilers/Core/Portable/Serialization/ObjectReader.cs

```diff
@@ -97,13 +97,24 @@ internal sealed partial class ObjectReader : IDisposable
                 return null;
             }
 
-            if (stream.ReadByte() != VersionByte1 ||
-                stream.ReadByte() != VersionByte2)
+            if (!tryReadVersionByte(stream, VersionByte1) ||
```

Comment:
You might want to take this commit: https://github.com/dotnet/roslyn/pull/47864/commits/8d66dcf411617eb8e47290f58bbfe1be98d009e7

Reply:
I'm going to revert my commit later since I'm just using it for diagnostics, but the one you referenced looks like a good direction.